From b059d28b324f651122f497ac36a55cf6a6532634 Mon Sep 17 00:00:00 2001
From: Chaitanya Kulkarni <kch@nvidia.com>
Date: Mon, 17 Jan 2022 00:31:30 -0800
Subject: [PATCH 1/3] nvmet: set blksize shift in to revalidate

In this preparation patch for file backed target namespace we move the
blksize shift calculation into nvmet_file_ns_relavalidate().
This is needed to implement using block device in file backend mode
so we can add block device so the blkshift calculation for block device
backend and for file backend can be kept in one revalidate helper.

Signed-off-by: Chaitanya Kulkarni <kch@nvidia.com>
---
 drivers/nvme/target/io-cmd-file.c | 15 ++++++++-------
 1 file changed, 8 insertions(+), 7 deletions(-)

diff --git a/drivers/nvme/target/io-cmd-file.c b/drivers/nvme/target/io-cmd-file.c
index 6be6e59d273b..633a32020747 100644
--- a/drivers/nvme/target/io-cmd-file.c
+++ b/drivers/nvme/target/io-cmd-file.c
@@ -23,6 +23,14 @@ int nvmet_file_ns_revalidate(struct nvmet_ns *ns)
 			  AT_STATX_FORCE_SYNC);
 	if (!ret)
 		ns->size = stat.size;
+
+	/*
+	 * i_blkbits can be greater than the universally accepted upper bound,
+	 * so make sure we export a sane namespace lba_shift.
+	 */
+	ns->blksize_shift = min_t(u8,
+			file_inode(ns->file)->i_blkbits, 12);
+
 	return ret;
 }
 
@@ -61,13 +69,6 @@ int nvmet_file_ns_enable(struct nvmet_ns *ns)
 	if (ret)
 		goto err;
 
-	/*
-	 * i_blkbits can be greater than the universally accepted upper bound,
-	 * so make sure we export a sane namespace lba_shift.
-	 */
-	ns->blksize_shift = min_t(u8,
-			file_inode(ns->file)->i_blkbits, 12);
-
 	ns->bvec_cache = kmem_cache_create("nvmet-bvec",
 			NVMET_MAX_MPOOL_BVEC * sizeof(struct bio_vec),
 			0, SLAB_HWCACHE_ALIGN, NULL);
-- 
2.29.0

